src/nvim/memline.c

```diff
@@ -3100,24 +3013,21 @@ char_u *makeswapname(char_u *fname, char_u *ffname, buf_T *buf, char_u *dir_name
 /// @param dname  don't use "dirname", it is a global for Alpha
 char *get_file_in_dir(char *fname, char *dname)
 {
-  char *t;
-  char *tail;
   char *retval;
-  int save_char;
 
-  tail = path_tail(fname);
+  char *tail = path_tail((char *)fname);
 
   if (dname[0] == '.' && dname[1] == NUL) {
     retval = xstrdup(fname);
   } else if (dname[0] == '.' && vim_ispathsep(dname[1])) {
     if (tail == fname) {            // no path before file name
       retval = concat_fnames(dname + 2, tail, true);
     } else {
-      save_char = (uint8_t)(*tail);
+      int save_char = *tail;
```

Comment:
Better change this to `char` change remove the cast below.
```suggestion
      char save_char = *tail;
```